pandas/tests/groupby/test_groupby.py

```diff
@@ -2419,3 +2419,20 @@ def test_rolling_wrong_param_min_period():
     result_error_msg = r"__init__\(\) got an unexpected keyword argument 'min_period'"
     with pytest.raises(TypeError, match=result_error_msg):
         test_df.groupby("name")["val"].rolling(window=2, min_period=1).sum()
+
+
+@pytest.mark.parametrize(
+    "func, expected",
+    [("sum", [5, 7, 9]), ("std", [4.5 ** 0.5] * 3), ("var", [4.5] * 3)],
+)
+@pytest.mark.parametrize("dtype", [int, "Int64"])
+def test_multiindex_groupby(dtype, func, expected):
+    # GH#43209
+    df = DataFrame(
+        [[1, 2, 3, 4, 5, 6]] * 3,
+        columns=MultiIndex.from_product([["a", "b"], ["i", "j", "k"]]),
+    ).astype({("a", "j"): dtype, ("b", "j"): dtype})
+    result = df.groupby(level=1, axis=1).agg(func)
+    out_dtype = dtype if func == "sum" else None
+    expected = DataFrame([expected] * 3, columns=["i", "j", "k"], dtype=out_dtype)
+    tm.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
the entire point is to have `check_dtype=True` (the default)

Reply:
Hi @jreback - removed that.. let me know how this looks